### PR TITLE
Update forecast job to use COMIN/COMOUT

### PR DESCRIPTION
**Pull request by WalterKolczynski-NOAA**
# Description
NCO has requested that each COM variable specify whether it is an input or an output. This completes that process for the forecast job.

Refs #2451

# Type of change
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
- hybrid DA test on Hercules
- GEFS test on Hercules

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Review by aerorahul — APPROVED** [review]
looks good.

File: ush/forecast_postdet.sh [hunk 1]
```diff
@@ -583,7 +583,7 @@ CICE_out() {
   echo "SUB ${FUNCNAME[0]}: Copying output data for CICE"
 
   # Copy ice_in namelist from DATA to COMOUTice after the forecast is run (and successfull)
```
Comment (DavidHuber-NOAA) [inline ush/forecast_postdet.sh]: Should probably update this comment
```suggestion
  # Copy ice_in namelist from DATA to COMOUT_CONF after the forecast is run (and successfull)
```

**Review by KateFriedman-NOAA — APPROVED** [review]
Looks good, thanks @WalterKolczynski-NOAA !

**Review by DavidHuber-NOAA — APPROVED** [review]
Looks good.  Just one suggestion regarding a comment.

**Pull request merged**